webui/react/src/routes/utils.ts

```diff
@@ -206,6 +206,8 @@ export const handlePath = (
 
   const href = options.path ? linkPath(options.path, options.external) : undefined;
 
+  if (href === window.location.pathname) return; // avoid runing all logic again if clicked on a link to the current location
```

Comment:
if it safe to add this code?  no implications? cuz this function is used in other places